### ISSUE TITLE
The prediction accuracy began to hover around 60%, and even slowly declined

### ISSUE THREAD

**Issue by hemo0826** [binli123/dsmil-wsi#46]
<img width="921" alt="image" src="https://user-images.githubusercontent.com/83693017/175454602-f1230020-d73d-4b99-b1b5-083127a66541.png">

<img width="847" alt="image" src="https://user-images.githubusercontent.com/83693017/175470770-dde9da38-2efd-48ed-9ab8-72de7a83c8a9.png">

Hello, I tested on my own binary data set. After simclr (batch_size=256) feature extraction, I found that my training accuracy has been hovering around 60%, and finally it will even be predicted that it is all in the same category. Is it me_ Feats error or something else, and bag_ The prediction values are very close, both in the 0.1 attachment .please point it out, thank you

**Comment by dawalisi** [comment]
hello, has this problem been solved? I also have the same problem

**Comment by binli123** [comment]
#44 Maybe these two threads should be put together?

**Comment by binli123** [comment]
I incorporated the training/testing into the same pipeline in the latest commit. You can set --eval_scheme=5-fold-cv-standalone-test which will perform a train/valid/test like this: 
>A standalone test set consisting of 20% samples is reserved, remaining 80% of samples are used to construct a 5-fold cross-validation.
>For each fold, the best model and corresponding threshold are saved.
>After the 5-fold cross-validation, 5 best models along with the corresponding optimal thresholds are obtained which are used to perform inference on the reserved test set. A final prediction for a test sample is the majority vote of the 5 models.
>For a binary classification, accuracy and balanced accuracy scores are computed. For a multi-label classification, hamming loss (smaller the better) and subset accuracy are computed.

You can also simply run a 5-fold cv --eval_scheme=5-fold-cv

There were some issues with the testing script when loading pretrained weights (i.e., sometimes the weights are not fully loaded or there are missing weights, setting strict=False can reveal the problems.). The purpose of the testing script is to generate the heatmap, you should now read the performance directly from the training script. I will fix the issues in a couple of days.
